modin/core/dataframe/algebra/default2pandas/groupby.py

```diff
@@ -236,13 +241,13 @@ def fn(
             as_index = groupby_kwargs.pop("as_index", True)
             groupby_kwargs["as_index"] = True
 
-            grp = df.groupby(by, axis=axis, **groupby_kwargs)
+            grp = cls._call_groupby(df, by, axis=axis, **groupby_kwargs)
             func = cls.get_func(agg_func, **kwargs)
             result = func(grp, *agg_args, **agg_kwargs)
             method = kwargs.get("method")
 
             if isinstance(result, pandas.Series):
-                result = result.to_frame(MODIN_UNNAMED_SERIES_LABEL)
+                result = result.to_frame(result.name or MODIN_UNNAMED_SERIES_LABEL)
```

Comment:
let's make this check explicit, otherwise, it would work incorrectly for '0' name or empty name
```suggestion
                result = result.to_frame(MODIN_UNNAMED_SERIES_LABEL if result.name is None else result.name)
```

Reply:
agreed